### ISSUE TITLE
MultKan and KAN

### ISSUE THREAD

**Issue by FulingChen** [KindXiaoming/pykan#380]
I used the original version of KAN on my data which works perfectly, default settings with width [10,1]. Later on, I updated to the MultiKAN version. I just tried it on the same data with same settings, but the model couldn't converge. I wonder is there any algorithm difference between MultKan and KAN, despite the adding in multiplication? I assume if not using multiplication, MultiKan should just be KAN.

**Comment by FulingChen** [comment]
Here is from the last version of my code
model = KAN(width=[1347,10,1], grid=3, k=3, seed=0, device=device)
model.train(
            dataset,
            steps=100,
            lamb=0.1,
            lamb_entropy=2,
            lamb_l1=2,
            device=device,
        )

I have 1347 variants and just need one layer of 10 neurons.
I didn't save the log of the old model, but it converges after 3-5 steps.

Here is the code using the latest version.
model = KAN(
            width=[1347,10,1],
            grid=3,
            k=3,
            seed=0,
            ckpt_path=ckpt_path,
            device=device,)
model.fit(
            dataset,
            steps=100,
            lamb=0.1,
            lamb_entropy=2,
            lamb_l1=2,
            # lr=lr,
        )
It starts with a reasonable loss, but after 5-6 steps, the loss is nan.

**Comment by KindXiaoming** [comment]
based on my experience, lamb=0.1 might be too large, please try smaller lamb or try fit(..., update_grid=False)

**Comment by FulingChen** [comment]
Thank you so much. False update_grid helps me solving the issue. Just wondering why.
